persist/fs/persist_manager.go

```diff
@@ -230,15 +233,30 @@ func (pm *persistManager) Prepare(
 		return prepared, errPersistManagerCannotPrepareNotFlushing
 	}
 
-	// NB(xichen): if the checkpoint file for blockStart already exists, bail.
-	// This allows us to retry failed flushing attempts because they wouldn't
-	// have created the checkpoint file.
-	if FilesetExistsAt(pm.filePathPrefix, nsID, shard, blockStart) {
-		return prepared, nil
+	// If the checkpoint file aleady exists, bail. This allows us to retry failed attempts because
+	// they wouldn't have created the checkpoint file. This can happen in a variety of situations
+	// for flushes, but is unlikely with snapshots because every snapshot has a unique timestamp
+	// that is not block-aligned.
+	if opts.IsSnapshot {
```

Comment:
maybe put methods on the `opts` to simplify this logic:

```
func (o *PrepareOptions) FilesetExistsAt(prefix string) bool {
  if o.IsSnapshot() { return SnapshotFilesetExistsAt(prefix, ...) }
  return DataFilesetExistsAt(prefix, ...)
}
```

Reply:
ehh something about putting that method on the options struct seems gross. I'll make it a helper on the persist manager

Reply:
done